lib/commands/publish/npmUpdateAsLatest.js

```diff
@@ -12,13 +12,19 @@ module.exports = function npmUpdateAsLatest(changedPackages, versions, callback)
     return function (done) {
       while (true) {
         try {
-          logger.log("info", "Removing temporary npm dist-tag");
-          npmUtils.removeDistTag(pkg.name, "lerna-temp");
+          if (npmUtils.checkDistTag(pkg.name, "lerna-temp")) {
+            logger.log("info", "Removing temporary npm dist-tag");
+            npmUtils.removeDistTag(pkg.name, "lerna-temp");
+          }
+
           if (process.env.NPM_DIST_TAG) {
             logger.log("info", "Adding " + process.env.NPM_DIST_TAG + " npm dist-tag");
             npmUtils.addDistTag(pkg.name, versions[pkg.name], process.env.NPM_DIST_TAG);
+          } else if (flags.canary) {
+            logger.log("info", "Adding canary npm dist-tag");
+            npmUtils.addDistTag(pkg.name, pkg.pkg.version, "canary");
```

Comment:
https://github.com/hzoo/lerna/blob/publish-dev/lib/commands/publish/updateChangedPackages.js#L46-L58

in `updateChangedPackages` we used to only change the updated packages.

I changed it to run on all packages, and then set the version in `pkg.pkg.version = version;` at the end we rewrite the package.json